src/gif.rs

```diff
@@ -123,8 +141,9 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for GifDecoder<R> {
                 // See the comments inside `<GifFrameIterator as Iterator>::next` about
                 // the error handling of `from_raw`.
                 let image = ImageBuffer::from_raw(f_width, f_height, &mut *buf).ok_or_else(
-                    || ImageError::UnsupportedError("Image dimensions are too large".into())
-                )?;
+                    || ImageError::Decoding(DecodingError::new(
```

Comment:
This is not a decoding error. It means that the width and height are within specification but the platform we run on can not support the buffer space we would require.

Reply:
It seems I missed this in `bmp`.

Reply:
Fixed, will do the same to `bmp`